### ISSUE TITLE
When I fine-tune model, Do i need to modify "convert_char_to_pinyin" by other function ?

### ISSUE THREAD

**Issue by PhamDangNguyen** [SWivid/F5-TTS#263]
Chinese is a unique language. When I want to train and process with other languages, such as Vietnamese, Spanish, or Lao, do I need to change the data processing method? Specifically, the function 'convert_char_to_pinyin'

**Comment by vu-the-dung** [comment]
I only tested with Vietnamese. I saw that after adding missing Vietnamese characters to vocab.txt, when calling convert_char_to_pinyin with a sentence in Vietnamese, the returned char array often have extra space character before those added characters. 
So I changed convert_char_to_pinyin (in both prepare_csv_wavs.py and utils_infer.py) to a function that only convert string to char array, run prepare_csv_wavs.py to preprocess my dataset, then train/finetune.

So far I've finetuned my ~300h dataset based on the F5TTS_Base checkpoint, currently at 1m5 steps, it can speak intelligible Vietnamese, but still have issue of speaking repeated words, missing words, wrong word order

**Comment by vu-the-dung** [comment]
@PhamDangNguyen my dataset already have normalized text, even without punctuation. And after some test inferences with punctuation in gen_text, I'm feeling like the model have forgotten to pause at those punctuation. I think you can try to keep punctuation in your dataset's text

**Comment by PhamDangNguyen** [comment]
@vu-the-dung please give me some example, i'm newbie in TTS

**Comment by PhamDangNguyen** [comment]
@vu-the-dung I think the data needs additional periods and commas because the model will learn to pause accordingly. Also, capital letters will help the model learn to emphasize tone in the audio. What do you think? Worth a try!

**Comment by PhamDangNguyen** [comment]
@vu-the-dung, oh no problem. But i think the data needs additional periods and commas because the model will learn to pause accordingly. Also, capital letters will help the model learn to emphasize tone in the audio. What do you think? Worth a try!

**Comment by TheMattBin** [comment]
Hi @vu-the-dung, may I know that what have you changed in `convert_char_to_pinyin` you mentioned as I'm trying to train Cantonese. Thanks!

**Comment by vu-the-dung** [comment]
> Hi @vu-the-dung, may I know that what have you changed in `convert_char_to_pinyin` you mentioned as I'm trying to train Cantonese. Thanks!

This is what i use to replace `convert_char_to_pinyin`, i define it in `src/f5_tts/model/utils.py`
`def convert_string_to_char_list(text_list):
    final_text_list = []
    for text in text_list:
        char_list = [char for char in text if char not in "。，、；：？！《》【】—…:;\"()[]{}"]
        final_text_list.append(char_list)
    return final_text_list`

Since Vienamese alphabet is mainly Latin characters, with the addition of vowels with accent marks (eg: 'á', 'ắ', 'ồ', 'ự',...), i simply convert the text string to characters array.

As i know, Cantonese use Chinese Characters. I don't know about what's different between romanized Mandarin and romanized Cantonese characters. Maybe @SWivid can help?

**Comment by SWivid** [comment]
@vu-the-dung no idea lol, will have difference (thought quite a lot) though I can't speak Cantonese.

**Comment by TheMattBin** [comment]
@vu-the-dung Thanks for the reply! So basically you're doing sort of extension with addition of vowels with accent marks?
@SWivid for text should be kind similar, just simplified and traditional. I will check words with gradio UI. Thanks!

**Comment by indiejoseph** [comment]
For Cantonese, I think you can follow what we did with GPT-SoVITs, we added a g2p function for convert Chinese word to phoneme, which Jyutping a romanization for Cantonese. https://github.com/hon9kon9ize/GPT-SoVITS-Cantonese/blob/v2_(wip)/GPT_SoVITS/text/g2pw/g2pw.py

**Comment by TheMattBin** [comment]
> For Cantonese, I think you can follow what we did with GPT-SoVITs, we added a g2p function for convert Chinese word to phoneme, which Jyutping a romanization for Cantonese. https://github.com/hon9kon9ize/GPT-SoVITS-Cantonese/blob/v2_(wip)/GPT_SoVITS/text/g2pw/g2pw.py

Thanks so much! I will definitely take a look, I also try GPT-SoVITs before. However the voice cloned is quite robotic to me.

**Comment by indiejoseph** [comment]
Oh sorry, this is our private repo, Bert-Vits2 version instead https://github.com/hon9kon9ize/Bert-VITS2-Cantonese/blob/main/text/cantonese.py
